clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/deploy/DcosServerGroupNameResolver.groovy

```diff
@@ -36,19 +36,20 @@ class DcosServerGroupNameResolver extends AbstractServerGroupNameResolver {
     List<App> apps = appNamespaceResponse != null && appNamespaceResponse.isPresent() ? appNamespaceResponse.get().apps : []
 
     if (!apps) {
-      return Collections.emptyList()
+      return []
     }
 
     def filteredApps = apps.findAll {
-      DcosSpinnakerAppId.parse(it.id, true).get().serverGroupName.cluster == Names.parseName(clusterName).cluster
+      def appId = DcosSpinnakerAppId.parseVerbose(it.id)
+      appId.isPresent() && appId.get().namespace == region && appId.get().serverGroupName.cluster == Names.parseName(clusterName).cluster
     }
 
     return filteredApps.collect { App app ->
-      final def names = DcosSpinnakerAppId.parse(app.id, true).get().serverGroupName
+      final def names = DcosSpinnakerAppId.parseVerbose(app.id).get().serverGroupName
       return new AbstractServerGroupNameResolver.TakenSlot(
         serverGroupName: names.cluster,
         sequence       : names.sequence,
-        createdTime    : new Date(translateTime(app.versionInfo.lastConfigChangeAt))
+        createdTime    : new Date(translateTime(app.version))
```

Comment:
Don't think that's necessary but just wanted to make sure that got updated on the jira since I wasn't aware that was getting done.